### PR TITLE
Support underscore syntax in numbers #7181 (rax2)

### PR DESCRIPTION
**Pull request by SajjadPourali**
#7181
It's just work on hex.
usage : rax2 0xaaaa_aaab_dada.
if you agree with this i will add it on the other types .

### CODE REVIEW

File: binr/rax2/rax2.c [hunk 1]
```diff
@@ -422,6 +433,7 @@ static int rax(char *str, int len, int last) {
 	}
 
 	if (str[0] == '0' && str[1] == 'x') {
+		remove_seperator(str,2,'_',4);
```
Comment (alvarofe) [inline binr/rax2/rax2.c]: no space before ( and between ,

**Comment by radare** [comment]
imho this shuold be implemented in r_num, not rax2

also, there’s nt much use of underscore with hexa, its usually used for decimal base to separate every 3 digits like 1_000_000.

**Comment by alvarofe** [comment]
But why support underscore when we have dots (1.000.000)? is the standard way to express it,  isn't?

**Comment by SajjadPourali** [comment]
@radare : i saw that's label, #7181 label is rax2 .
anyway if you are agree with this i will add this to decimal and binary and other types (like this https://www.python.org/dev/peps/pep-0515/#abstract-and-rationale).

**Comment by SajjadPourali** [comment]
@alvarofe : I'm agree with you, i just implement something that in #7181.
i think it's better to remove any symbols from input (so users can uses any char for separator).

**Comment by radare** [comment]
in europe dot is fine, but in US they use a comma, and programming languages use underscore to avoid confussion with floating point numbers

**Comment by radare** [comment]
if you implement this into r_num , it will work on all r2, not just rax2

**Comment by SajjadPourali** [comment]
@radare : Ok, I will implement it into r_num .

File: binr/rax2/rax2.c [hunk 2]
```diff
@@ -71,6 +71,17 @@ static int format_output(char mode, const char *s) {
 	return true;
 }
 
+static void separator_remover(char * str, int pad, char symbol, int period) {
```
Comment (radare) [inline binr/rax2/rax2.c]: r_str_replace_char(str, '_', 0); does the same but faster and more generic

**Comment by radare** [comment]
Done in https://github.com/radare/radare2/pull/7896

**Pull request closed without merging**